src/main.js

```diff
@@ -37,29 +38,82 @@ document.addEventListener("DOMContentLoaded", function(event) {
     if(initialLoad) {
       initialLoad = false;
       window.easymde.codemirror.getDoc().clearHistory();
-      if(window.easymde.isPreviewActive()) {
+      var mode = clientData && clientData.mode;
+
+      // Set initial editor mode
+      if(mode === 'preview') {
+        if(!window.easymde.isPreviewActive()) {
+          window.easymde.togglePreview();
+        }
+      } else if(mode === 'split') {
+        if(!window.easymde.isSideBySideActive()) {
+          window.easymde.toggleSideBySide();
+        }
+      // falback config
+      } else if(window.easymde.isPreviewActive()) {
         window.easymde.togglePreview();
       }
     }
   });
 
   window.easymde = new EasyMDE({
-     element: document.getElementById("editor"),
-     spellChecker: false,
-     status: false,
-     shortcuts: {
-       toggleSideBySide: "Cmd-Alt-P"
-     },
-     toolbar:[
-           "heading", "bold", "italic", "strikethrough",
-           "|", "quote", "code",
-           "|", "unordered-list", "ordered-list",
-           "|", "clean-block",
-           "|", "link", "image",
-           "|", "table",
-           "|", "preview", "side-by-side"
-           ],
-   });
+    element: document.getElementById("editor"),
+    spellChecker: false,
+    status: false,
+    shortcuts: {
+      toggleSideBySide: "Cmd-Alt-P"
+    },
+    toolbar:[
+      "heading", "bold", "italic", "strikethrough",
+      "|", "quote", "code",
+      "|", "unordered-list", "ordered-list",
+      "|", "clean-block",
+      "|", "link", "image",
+      "|", "table",
+      "|",
+      {
+        className: "fa fa-eye",
+        default: true,
+        name: "preview",
+        noDisable: true,
+        title: "Toggle Preview",
+        action: function() {
+          window.easymde.togglePreview();
+          saveMetadata();
+        }
+      },
+      {
+        className: "fa fa-columns",
+        default: true,
+        name: "side-by-side",
+        noDisable: true,
+        noMobile: true,
```

Comment:
Why `noMobile` true?

Reply:
Also, would it have been better to add a click listener on the existing buttons instead of redefining them?

Reply:
I simply applied the default configs EasyMDE [uses](https://github.com/Ionaru/easy-markdown-editor/blob/master/src/js/easymde.js#L1305) for the `side-by-side` toolbar command since that's what we're using.

Reply:
Yeah, I wanted to do that too but I couldn't find a way to add the `action` to an already existing button. From the [documentation](https://github.com/Ionaru/easy-markdown-editor#toolbar-customization) it seems you either use a predefined button or have to define everything yourself 😞